src/current/advisories/a104309.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Technical Advisory a104309
+advisory: A-a104309
+summary: In rare cases, a rangefeed bug may cause a checkpoint to be emitted prematurely, before all writes below the checkpoint timestamp have been emitted.
+toc: true
+affected_versions: v2.1.11 to v22.2.17, v23.1.0 to v23.1.14, v23.2.0
+advisory_date: 2024-02-20
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+A [rangefeed](https://www.cockroachlabs.com/docs/stable/create-and-configure-changefeeds#enable-rangefeeds) bug may cause a [checkpoint](https://www.cockroachlabs.com/docs/stable/how-does-an-enterprise-changefeed-work) to be emitted prematurely in rare cases, before all writes below the checkpoint timestamp have been emitted. This could occur if nodes are overloaded or if long-running transactions (several seconds) are involved, such that intent resolution only replicates to a follower running a rangefeed more than 10 seconds after the transaction began. If a cluster using changefeeds experiences this bug, changefeeds will omit these write events, and the following error will be logged:
+
+{% include_cached copy-clipboard.html %}
+~~~ none
+cdc ux violation: detected timestamp ... that is less or equal to the local frontier
+~~~
+
+Across a large sample of clusters using changefeeds over a month, 3.8% of clusters had experienced this bug, and a total 0.000064% of events were omitted (one in a million).
+
+The bug is caused by a race condition that is typically triggered by transactions that run for several seconds or by follower replicas that lag significantly behind the leaseholder, such as when a node is overloaded or a network is disrupted. The rangefeed must be running on a follower replica that applies intent resolution long after the original transaction has committed and garbage-collected its intents and transaction record on the leaseholders (at least 10 seconds after the transaction begins).
+
+The specific conditions to trigger the bug are:
+
+1. An explicit or cross-range transaction commits, then asynchronously resolves all of its intents and removes its transaction record on all relevant leaseholders.
+
+1. A follower replica running a rangefeed replicates the transaction's intent writes, but not yet the resolution of the intents. More than 10 seconds must elapse between when the transaction begins and when the intent resolution is replicated to the follower. (The intent resolution must be replicated to the follower after 10 seconds. That means that the sum of `transaction duration + intent resolution duration + transaction record GC + replication to follower` must be greater than 10 seconds to trigger the bug.)
```

Comment:
This sentence seem unnecessary, if not redundant -- "The intent resolution must be replicated to the follower after 10 seconds."